sdk/src/client/api/block_builder/transaction_builder/context_inputs.rs

```diff
@@ -11,51 +11,67 @@ use crate::{
-                    if account.features().block_issuer().is_some() {
-                        log::debug!("Adding block issuance context input for transitioned account output");
-                        self.bic_context_inputs.insert(BlockIssuanceCreditContextInput::from(
-                            account.account_id_non_null(input.output_id()),
-                        ));
-                    }
+        match &input.output {
+            // Transitioning an issuer account requires a BlockIssuanceCreditContextInput.
+            Output::Account(account) => {
+                if account.features().block_issuer().is_some() {
+                    log::debug!("Adding block issuance context input for transitioned account output");
+                    self.bic_context_inputs.insert(BlockIssuanceCreditContextInput::from(
+                        account.account_id_non_null(input.output_id()),
+                    ));
                 }
-                // Transitioning an implicit account requires a BlockIssuanceCreditContextInput.
-                Output::Basic(basic) => {
-                    if basic.is_implicit_account() {
-                        log::debug!("Adding block issuance context input for transitioned implicit account output");
-                        self.bic_context_inputs
-                            .insert(BlockIssuanceCreditContextInput::from(AccountId::from(
-                                input.output_id(),
-                            )));
-                    }
+            }
+            // Transitioning an implicit account requires a BlockIssuanceCreditContextInput.
+            Output::Basic(basic) => {
+                if basic.is_implicit_account() {
+                    log::debug!("Adding block issuance context input for transitioned implicit account output");
+                    self.bic_context_inputs
+                        .insert(BlockIssuanceCreditContextInput::from(AccountId::from(
+                            input.output_id(),
+                        )));
                 }
-                _ => (),
             }
+            _ => (),
+        }
 
-            // Inputs with timelock or expiration unlock condition require a CommitmentContextInput
-            if input
-                .output
-                .unlock_conditions()
-                .map_or(false, |u| u.iter().any(|u| u.is_timelock() || u.is_expiration()))
-            {
-                log::debug!("Adding commitment context input for timelocked or expiring output");
-                needs_commitment_context = true;
-            }
+        // Inputs with timelock or expiration unlock condition require a CommitmentContextInput
+        if input
+            .output
+            .unlock_conditions()
+            .map_or(false, |u| u.iter().any(|u| u.is_timelock() || u.is_expiration()))
+        {
+            log::debug!("Adding commitment context input for timelocked or expiring output");
+            needs_commitment_context = true;
```

Comment:
Could add a check above all of these to skip these if there's already a commitment context input and remove the check at the bottom

Reply:
Thanks